### PR TITLE
fix  java.lang.IncompatibleClassChangeError when submit with spark-2.4.3

### PR DESCRIPTION
**Pull request by marsishandsome**
### What problem does this PR solve? 

close https://github.com/pingcap/tispark/issues/971

tispark throws java.lang.IncompatibleClassChangeError when submit with spark-2.4.3

the reason is that:
1. tispark is compiled with spark-2.3.3
2. but running with spark-2.4.3
3. some scala code in spark although is compatible between spark-2.3.3 and spark-2.4.3
4. but it's not compatible on the level of jvm byte code

### What is changed and how it works?
use reflection and call the incompatible code in spark-wrapper-2.3 and spark-wrapper-2.4

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/sql/extensions/rules.scala [hunk 1]
```diff
@@ -57,7 +58,7 @@ case class TiResolutionRule(getOrCreateTiContext: SparkSession => TiContext)(
       )(sqlContext)
       // Use SubqueryAlias so that projects and joins can correctly resolve
       // UnresolvedAttributes in JoinConditions, Projects, Filters, etc.
-      SubqueryAlias(tableName, LogicalRelation(tiDBRelation))
+      newSubqueryAlias(tableName, LogicalRelation(tiDBRelation))
```
Comment (zhexuany) [inline core/src/main/scala/org/apache/spark/sql/extensions/rules.scala]: How about change prefix `new` to `Ti`?

Comment (marsishandsome) [reply core/src/main/scala/org/apache/spark/sql/extensions/rules.scala]: but function name should start with `lowercase`? any better suggestions? maybe `tiSubqueryAlias`?

**Comment by birdstorm** [comment]
After discussion, we might need to move our tests to a new project to test its stability when depending on different spark version for the same assembly jar.

**Comment by marsishandsome** [comment]
> After discussion, we might need to move our tests to a new project to test its stability when depending on different spark version for the same assembly jar.

see https://github.com/pingcap/tispark/issues/974

File: core/src/main/scala/org/apache/spark/sql/extensions/rules.scala (hunk 1 above)
Comment (zhexuany) [reply core/src/main/scala/org/apache/spark/sql/extensions/rules.scala]: Sounds pretty nice

**Pull request merged**